hibernate-core/src/main/java/org/hibernate/type/TypeFactory.java

```diff
@@ -88,7 +92,24 @@ public Type byClass(Class clazz, Properties parameters) {
 
 	public Type type(Class<Type> typeClass, Properties parameters) {
 		try {
-			Type type = typeClass.newInstance();
+			Type type;
+
+			Constructor<Type> bootstrapContextAwareTypeConstructor = ReflectHelper.getConstructor(
+					typeClass,
+					TypeBootstrapContext.class
+			);
+			if ( bootstrapContextAwareTypeConstructor != null ) {
+				ConfigurationService configurationService = typeConfiguration.getServiceRegistry().getService(
+						ConfigurationService.class );
+				Map<String, Object> configurationSettings = configurationService.getSettings();
+				type = bootstrapContextAwareTypeConstructor.newInstance( new TypeBootstrapContext(
```

Comment:
Not a fan of creating a new `TypeBootstrapContext` every time.  I'd pass it the `TypeConfiguration` and be done with it, or have `TypeConfiguration` implement `TypeBootstrapContext` (after making it an interface - which it ought to be anyway imo)

Reply:
Or better yet, have TypeFactory be the TypeBootstrapContext

Reply:
I can work on this change

Reply:
Thanks.